src/sbvr-api/permissions.ts

```diff
@@ -387,12 +391,56 @@ const $checkPermissions = (
 	});
 };
 
-const constrainedPermissionError = new PermissionError();
-const generateConstrainedAbstractSql = (
+const convertToLamda = (filter: AnyObject, identifier: string) => {
+	const replaceObject = (object: AnyObject) => {
+		if (typeof object === 'string') {
+			return;
+		}
+		if (_.isArray(object)) {
+			object.forEach(element => {
+				replaceObject(element);
+			});
+		}
+
+		if (object.hasOwnProperty('name')) {
+			object.property = _.cloneDeep(object);
+			object.name = identifier;
+			delete object.lambda;
+		}
+	};
+
+	replaceObject(filter);
+};
+
+const rewriteSubPermissionBindings = (filter: AnyObject, counter: number) => {
```

Comment:
Isn't this equivalent to `rewriteBinds` in odata-to-abstract-sql?

Reply:
I think the `rewriteBinds` in `odata-to-abstract-sql` operates on the abstract sql representation, where this operates on the parsed odata representation. So this happens on the odata level before transforming it into abstract sql.